File: C++/599. Minimum Index Sum of Two Lists.cpp
class Solution {
public:
    vector<string> findRestaurant(vector<string>& list1, vector<string>& list2) {
        unordered_map<string, int>maple;
        vector<string> ans;
        int res = INT_MAX;
        for(int i=0;i<list1.size();i++)
            maple[list1[i]] = i;
        for(int i=0;i<list2.size();i++)
        {
            if(maple.find(list2[i]) != maple.end())
            {
                if(i + maple[list2[i]] < res)
                {
                    res = i + maple[list2[i]];
                    ans.clear();
                    ans.push_back(list2[i]);
                }
                else if(i + maple[list2[i]] == res)
                    ans.push_back(list2[i]);
            }
        }
        return ans;
    }
};

File: Python/599. Minimum Index Sum of Two Lists.py
class Solution:
    def findRestaurant(self, list1, list2):
        """
        :type list1: List[str]
        :type list2: List[str]
        :rtype: List[str]
        """
        ans, maple = list(), dict()
        res = float('inf')
        for i in range(len(list1)):
            maple[list1[i]] = i
        for i in range(len(list2)):
            if maple.get(list2[i]) != None:
                if i + maple[list2[i]] < res:
                    res = i + maple[list2[i]]
                    ans = list()
                    ans.append(list2[i])
                elif i + maple[list2[i]] == res:
                    ans.append(list2[i])
        return ans